### ISSUE TITLE
Policy not updating properly while adding GSI

### ISSUE THREAD

**Issue by jtushar53** [aws-amplify/amplify-cli#2375]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
When we add GSI, policies are not added properly resulting in manually adding table/Mytable/index/* to all tables from IAM console  
**To Reproduce**
1. amplify add storage
2. add GSI 
3. amplify push
4. call dynamo db query inside lambda using gsi

error
 is not authorized to perform: dynamodb:Query on resource: arn:aws:dynamodb:us-east-1:XXXXXXXXXXX:table/mytable-dev/index/myindexname

**Expected behavior**
It should add GSI permission in role for each added / updated GSI from the command line
**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**
My template 
```json
{
	"AWSTemplateFormatVersion": "2010-09-09",
	"Description": "Lambda resource stack creation using Amplify CLI",
	"Parameters": {
		"env": {
			"Type": "String"
		},
		"storageordersName": {
			"Type": "String",
			"Default": "storageordersName"
		},
		"storageordersArn": {
			"Type": "String",
			"Default": "storageordersArn"
		},
		"storagesitesName": {
			"Type": "String",
			"Default": "storagesitesName"
		},
		"storagesitesArn": {
			"Type": "String",
			"Default": "storagesitesArn"
		},
		"authcwm2a58cbebUserPoolId": {
			"Type": "String",
			"Default": "authcwm2a58cbebUserPoolId"
		},
		"storagejobsName": {
			"Type": "String",
			"Default": "storagejobsName"
		},
		"storagejobsArn": {
			"Type": "String",
			"Default": "storagejobsArn"
		}
	},
	"Conditions": {
		"ShouldNotCreateEnvResources": {
			"Fn::Equals": [
				{
					"Ref": "env"
				},
				"NONE"
			]
		}
	},
	"Resources": {
		"LambdaFunction": {
			"Type": "AWS::Lambda::Function",
			"Metadata": {
				"aws:asset:path": "./src",
				"aws:asset:property": "Code"
			},
			"Properties": {
				"Handler": "index.handler",
				"FunctionName": {
					"Fn::If": [
						"ShouldNotCreateEnvResources",
						"Master",
						{
							"Fn::Join": [
								"",
								[
									"Master",
									"-",
									{
										"Ref": "env"
									}
								]
							]
						}
					]
				},
				"Environment": {
					"Variables": {
						"ENV": {
							"Ref": "env"
						},
						"REGION": {
							"Ref": "AWS::Region"
						},
						"STORAGE_ORDERS_NAME": {
							"Ref": "storageordersName"
						},
						"STORAGE_ORDERS_ARN": {
							"Ref": "storageordersArn"
						},
						"STORAGE_SITES_NAME": {
							"Ref": "storagesitesName"
						},
						"STORAGE_SITES_ARN": {
							"Ref": "storagesitesArn"
						},
						"AUTH_CWM2A58CBEB_USERPOOLID": {
							"Ref": "authcwm2a58cbebUserPoolId"
						},
						"STORAGE_JOBS_NAME": {
							"Ref": "storagejobsName"
						},
						"STORAGE_JOBS_ARN": {
							"Ref": "storagejobsArn"
						}
					}
				},
				"Role": {
					"Fn::GetAtt": [
						"LambdaExecutionRole",
						"Arn"
					]
				},
				"Runtime": "nodejs10.x",
				"Timeout": "25",
				"Code": {
					"S3Bucket": "cwm-dev-20190718135044-deployment",
					"S3Key": "amplify-builds/MastersLambda-47666859347a6d726143-build.zip"
				}
			}
		},
		"LambdaExecutionRole": {
			"Type": "AWS::IAM::Role",
			"Properties": {
				"RoleName": {
					"Fn::If": [
						"ShouldNotCreateEnvResources",
						"cwmLambdaRoleb4c71c02",
						{
							"Fn::Join": [
								"",
								[
									"cwmLambdaRoleb4c71c02",
									"-",
									{
										"Ref": "env"
									}
								]
							]
						}
					]
				},
				"AssumeRolePolicyDocument": {
					"Version": "2012-10-17",
					"Statement": [
						{
							"Effect": "Allow",
							"Principal": {
								"Service": [
									"lambda.amazonaws.com"
								]
							},
							"Action": [
								"sts:AssumeRole"
							]
						}
					]
				}
			}
		},
		"lambdaexecutionpolicy": {
			"DependsOn": [
				"LambdaExecutionRole"
			],
			"Type": "AWS::IAM::Policy",
			"Properties": {
				"PolicyName": "lambda-execution-policy",
				"Roles": [
					{
						"Ref": "LambdaExecutionRole"
					}
				],
				"PolicyDocument": {
					"Version": "2012-10-17",
					"Statement": [
						{
							"Effect": "Allow",
							"Action": [
								"logs:CreateLogGroup",
								"logs:CreateLogStream",
								"logs:PutLogEvents"
							],
							"Resource": {
								"Fn::Sub": [
									"arn:aws:logs:${region}:${account}:log-group:/aws/lambda/${lambda}:log-stream:*",
									{
										"region": {
											"Ref": "AWS::Region"
										},
										"account": {
											"Ref": "AWS::AccountId"
										},
										"lambda": {
											"Ref": "LambdaFunction"
										}
									}
								]
							}
						},
						{
							"Effect": "Allow",
							"Action": [
								"dynamodb:GetItem",
								"dynamodb:Query",
								"dynamodb:Scan",
								"dynamodb:PutItem",
								"dynamodb:UpdateItem",
								"dynamodb:DeleteItem"
							],
							"Resource": [
								{
									"Ref": "storagejobsArn"
								},
								{
									"Fn::Join": [
										"/",
										[
											{
												"Ref": "storagejobsArn"
											},
											"index/*"
										]
									]
								}
							]
						}
					]
				}
			}
		},
		"AmplifyResourcesPolicy": {
			"DependsOn": [
				"LambdaExecutionRole"
			],
			"Type": "AWS::IAM::Policy",
			"Properties": {
				"PolicyName": "amplify-lambda-execution-policy",
				"Roles": [
					{
						"Ref": "LambdaExecutionRole"
					}
				],
				"PolicyDocument": {
					"Version": "2012-10-17",
					"Statement": [
						{
							"Effect": "Allow",
							"Action": [
								"dynamodb:Put*",
								"dynamodb:Create*",
								"dynamodb:BatchWriteItem",
								"dynamodb:Get*",
								"dynamodb:BatchGetItem",
								"dynamodb:List*",
								"dynamodb:Describe*",
								"dynamodb:Scan",
								"dynamodb:Query",
								"dynamodb:Update*",
								"dynamodb:RestoreTable*",
								"dynamodb:Delete*"
							],
							"Resource": [
								{
									"Ref": "storageordersArn"
								}
							]
						},
						{
							"Effect": "Allow",
							"Action": [
								"dynamodb:Put*",
								"dynamodb:Create*",
								"dynamodb:BatchWriteItem",
								"dynamodb:Get*",
								"dynamodb:BatchGetItem",
								"dynamodb:List*",
								"dynamodb:Describe*",
								"dynamodb:Scan",
								"dynamodb:Query",
								"dynamodb:Update*",
								"dynamodb:RestoreTable*",
								"dynamodb:Delete*"
							],
							"Resource": [
								{
									"Ref": "storagejobsArn"
								}
							]
						},
						{
							"Effect": "Allow",
							"Action": [
								"dynamodb:Put*",
								"dynamodb:Create*",
								"dynamodb:BatchWriteItem",
								"dynamodb:Get*",
								"dynamodb:BatchGetItem",
								"dynamodb:List*",
								"dynamodb:Describe*",
								"dynamodb:Scan",
								"dynamodb:Query",
								"dynamodb:Update*",
								"dynamodb:RestoreTable*",
								"dynamodb:Delete*"
							],
							"Resource": [
								{
									"Ref": "storagesitesArn"
								}
							]
						},
						{
							"Effect": "Allow",
							"Action": [
								"cognito-idp:ConfirmSignUp",
								"cognito-idp:AdminCreateUser",
								"cognito-idp:CreateUserImportJob",
								"cognito-idp:AdminSetUserSettings",
								"cognito-idp:AdminLinkProviderForUser",
								"cognito-idp:CreateIdentityProvider",
								"cognito-idp:AdminConfirmSignUp",
								"cognito-idp:AdminDisableUser",
								"cognito-idp:AdminRemoveUserFromGroup",
								"cognito-idp:SetUserMFAPreference",
								"cognito-idp:SetUICustomization",
								"cognito-idp:SignUp",
								"cognito-idp:VerifyUserAttribute",
								"cognito-idp:SetRiskConfiguration",
								"cognito-idp:StartUserImportJob",
								"cognito-idp:AdminSetUserPassword",
								"cognito-idp:AssociateSoftwareToken",
								"cognito-idp:CreateResourceServer",
								"cognito-idp:RespondToAuthChallenge",
								"cognito-idp:CreateUserPoolClient",
								"cognito-idp:AdminUserGlobalSignOut",
								"cognito-idp:GlobalSignOut",
								"cognito-idp:AddCustomAttributes",
								"cognito-idp:CreateGroup",
								"cognito-idp:CreateUserPool",
								"cognito-idp:AdminForgetDevice",
								"cognito-idp:AdminAddUserToGroup",
								"cognito-idp:AdminRespondToAuthChallenge",
								"cognito-idp:ForgetDevice",
								"cognito-idp:CreateUserPoolDomain",
								"cognito-idp:AdminEnableUser",
								"cognito-idp:AdminUpdateDeviceStatus",
								"cognito-idp:StopUserImportJob",
								"cognito-idp:InitiateAuth",
								"cognito-idp:AdminInitiateAuth",
								"cognito-idp:AdminSetUserMFAPreference",
								"cognito-idp:ConfirmForgotPassword",
								"cognito-idp:SetUserSettings",
								"cognito-idp:VerifySoftwareToken",
								"cognito-idp:AdminDisableProviderForUser",
								"cognito-idp:SetUserPoolMfaConfig",
								"cognito-idp:ChangePassword",
								"cognito-idp:ConfirmDevice",
								"cognito-idp:AdminResetUserPassword",
								"cognito-idp:ResendConfirmationCode",
								"cognito-identity:Describe*",
								"cognito-identity:Get*",
								"cognito-identity:List*",
								"cognito-idp:Describe*",
								"cognito-idp:AdminGetDevice",
								"cognito-idp:AdminGetUser",
								"cognito-idp:AdminList*",
								"cognito-idp:List*",
								"cognito-sync:Describe*",
								"cognito-sync:Get*",
								"cognito-sync:List*",
								"iam:ListOpenIdConnectProviders",
								"iam:ListRoles",
								"sns:ListPlatformApplications",
								"cognito-idp:ForgotPassword",
								"cognito-idp:UpdateAuthEventFeedback",
								"cognito-idp:UpdateResourceServer",
								"cognito-idp:UpdateUserPoolClient",
								"cognito-idp:AdminUpdateUserAttributes",
								"cognito-idp:UpdateUserAttributes",
								"cognito-idp:UpdateUserPoolDomain",
								"cognito-idp:UpdateIdentityProvider",
								"cognito-idp:UpdateGroup",
								"cognito-idp:AdminUpdateAuthEventFeedback",
								"cognito-idp:UpdateDeviceStatus",
								"cognito-idp:UpdateUserPool",
								"cognito-idp:DeleteUserPoolDomain",
								"cognito-idp:DeleteResourceServer",
								"cognito-idp:DeleteGroup",
								"cognito-idp:AdminDeleteUserAttributes",
								"cognito-idp:DeleteUserPoolClient",
								"cognito-idp:DeleteUserAttributes",
								"cognito-idp:DeleteUserPool",
								"cognito-idp:AdminDeleteUser",
								"cognito-idp:DeleteIdentityProvider",
								"cognito-idp:DeleteUser"
							],
							"Resource": [
								{
									"Fn::Join": [
										"",
										[
											"arn:aws:cognito-idp:",
											{
												"Ref": "AWS::Region"
											},
											":",
											{
												"Ref": "AWS::AccountId"
											},
											":userpool/",
											{
												"Ref": "authcwm2a58cbebUserPoolId"
											}
										]
									]
								}
							]
						}
					]
				}
			}
		}
	},
	"Outputs": {
		"Name": {
			"Value": {
				"Ref": "LambdaFunction"
			}
		},
		"Arn": {
			"Value": {
				"Fn::GetAtt": [
					"LambdaFunction",
					"Arn"
				]
			}
		},
		"Region": {
			"Value": {
				"Ref": "AWS::Region"
			}
		},
		"LambdaExecutionRole": {
			"Value": {
				"Ref": "LambdaExecutionRole"
			}
		}
	}
}
```

My amplify-lambda-execution-policy
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Action": [
                "dynamodb:Put*",
                "dynamodb:Create*",
                "dynamodb:BatchWriteItem",
                "dynamodb:Get*",
                "dynamodb:BatchGetItem",
                "dynamodb:List*",
                "dynamodb:Describe*",
                "dynamodb:Scan",
                "dynamodb:Query",
                "dynamodb:Update*",
                "dynamodb:RestoreTable*",
                "dynamodb:Delete*"
            ],
            "Resource": [
                "arn:aws:dynamodb:us-east-1:219460060369:table/orders-dev"
            ],
            "Effect": "Allow"
        },
        {
            "Action": [
                "dynamodb:Put*",
                "dynamodb:Create*",
                "dynamodb:BatchWriteItem",
                "dynamodb:Get*",
                "dynamodb:BatchGetItem",
                "dynamodb:List*",
                "dynamodb:Describe*",
                "dynamodb:Scan",
                "dynamodb:Query",
                "dynamodb:Update*",
                "dynamodb:RestoreTable*",
                "dynamodb:Delete*"
            ],
            "Resource": [
                "arn:aws:dynamodb:us-east-1:219460060369:table/jobs-dev"
            ],
            "Effect": "Allow"
        },
        {
            "Action": [
                "dynamodb:Put*",
                "dynamodb:Create*",
                "dynamodb:BatchWriteItem",
                "dynamodb:Get*",
                "dynamodb:BatchGetItem",
                "dynamodb:List*",
                "dynamodb:Describe*",
                "dynamodb:Scan",
                "dynamodb:Query",
                "dynamodb:Update*",
                "dynamodb:RestoreTable*",
                "dynamodb:Delete*"
            ],
            "Resource": [
                "arn:aws:dynamodb:us-east-1:219460060369:table/sites-dev"
            ],
            "Effect": "Allow"
        },
        {
            "Action": [
                "cognito-idp:ConfirmSignUp",
                "cognito-idp:AdminCreateUser",
                "cognito-idp:CreateUserImportJob",
                "cognito-idp:AdminSetUserSettings",
                "cognito-idp:AdminLinkProviderForUser",
                "cognito-idp:CreateIdentityProvider",
                "cognito-idp:AdminConfirmSignUp",
                "cognito-idp:AdminDisableUser",
                "cognito-idp:AdminRemoveUserFromGroup",
                "cognito-idp:SetUserMFAPreference",
                "cognito-idp:SetUICustomization",
                "cognito-idp:SignUp",
                "cognito-idp:VerifyUserAttribute",
                "cognito-idp:SetRiskConfiguration",
                "cognito-idp:StartUserImportJob",
                "cognito-idp:AdminSetUserPassword",
                "cognito-idp:AssociateSoftwareToken",
                "cognito-idp:CreateResourceServer",
                "cognito-idp:RespondToAuthChallenge",
                "cognito-idp:CreateUserPoolClient",
                "cognito-idp:AdminUserGlobalSignOut",
                "cognito-idp:GlobalSignOut",
                "cognito-idp:AddCustomAttributes",
                "cognito-idp:CreateGroup",
                "cognito-idp:CreateUserPool",
                "cognito-idp:AdminForgetDevice",
                "cognito-idp:AdminAddUserToGroup",
                "cognito-idp:AdminRespondToAuthChallenge",
                "cognito-idp:ForgetDevice",
                "cognito-idp:CreateUserPoolDomain",
                "cognito-idp:AdminEnableUser",
                "cognito-idp:AdminUpdateDeviceStatus",
                "cognito-idp:StopUserImportJob",
                "cognito-idp:InitiateAuth",
                "cognito-idp:AdminInitiateAuth",
                "cognito-idp:AdminSetUserMFAPreference",
                "cognito-idp:ConfirmForgotPassword",
                "cognito-idp:SetUserSettings",
                "cognito-idp:VerifySoftwareToken",
                "cognito-idp:AdminDisableProviderForUser",
                "cognito-idp:SetUserPoolMfaConfig",
                "cognito-idp:ChangePassword",
                "cognito-idp:ConfirmDevice",
                "cognito-idp:AdminResetUserPassword",
                "cognito-idp:ResendConfirmationCode",
                "cognito-identity:Describe*",
                "cognito-identity:Get*",
                "cognito-identity:List*",
                "cognito-idp:Describe*",
                "cognito-idp:AdminGetDevice",
                "cognito-idp:AdminGetUser",
                "cognito-idp:AdminList*",
                "cognito-idp:List*",
                "cognito-sync:Describe*",
                "cognito-sync:Get*",
                "cognito-sync:List*",
                "iam:ListOpenIdConnectProviders",
                "iam:ListRoles",
                "sns:ListPlatformApplications",
                "cognito-idp:ForgotPassword",
                "cognito-idp:UpdateAuthEventFeedback",
                "cognito-idp:UpdateResourceServer",
                "cognito-idp:UpdateUserPoolClient",
                "cognito-idp:AdminUpdateUserAttributes",
                "cognito-idp:UpdateUserAttributes",
                "cognito-idp:UpdateUserPoolDomain",
                "cognito-idp:UpdateIdentityProvider",
                "cognito-idp:UpdateGroup",
                "cognito-idp:AdminUpdateAuthEventFeedback",
                "cognito-idp:UpdateDeviceStatus",
                "cognito-idp:UpdateUserPool",
                "cognito-idp:DeleteUserPoolDomain",
                "cognito-idp:DeleteResourceServer",
                "cognito-idp:DeleteGroup",
                "cognito-idp:AdminDeleteUserAttributes",
                "cognito-idp:DeleteUserPoolClient",
                "cognito-idp:DeleteUserAttributes",
                "cognito-idp:DeleteUserPool",
                "cognito-idp:AdminDeleteUser",
                "cognito-idp:DeleteIdentityProvider",
                "cognito-idp:DeleteUser"
            ],
            "Resource": [
                "arn:aws:cognito-idp:us-east-1:219460060369:userpool/us-east-1_wRsd6DaxE"
            ],
            "Effect": "Allow"
        }
    ]
}
```

**Comment by ammarkarachi** [comment]
@jtushar53 Was the GSI created after the Lambda?

**Comment by jtushar53** [comment]
GSI was created before the lambda, but i think the problem is not related to lambda, instead index/* is not added to storage's policy. So when i add storage and GSI and push it, i cannot find table/index/* in policy associated with storage

**Comment by stefangomez** [comment]
I'm seeing this too. I would attempt at making a PR for fixing but I don't know enough about the plugin architecture. The fix should be done somewhere around here I believe: https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-category-function/provider-utils/awscloudformation/service-walkthroughs/lambda-walkthrough.js#L733

Not sure if your labels matter, maybe for prioritization, but the reason I say it's a "bug" is because it used to work in previous versions of cli, and was fixed by: https://github.com/aws-amplify/amplify-cli/commit/cc2f1b66963b91e34169455e6dcdb04bb1cc9f87. 

luckily it's an easy (yet annoying) fix. @jtushar53 - assuming you fixed yourself, but if not, just add this to your lambda CF template under resource:

```json								
{
    "Fn::Join": [
        "/",
        [
            {
                "Ref": "#replace with your db arn parameter#"
            },
            "index/*"
        ]
    ]
}
```

So if your db arn param was `myappdbArn` your "Resource" block would look like this:

```json
"Resource": [
    {
        "Ref": "myappdbArn"
    },
    {
        "Fn::Join": [
            "/",
            [
                {
                    "Ref": "myappdbArn"
                },
                "index/*"
            ]
        ]
    }
]
```

**Comment by kaustavghosh06** [comment]
Fix for this was merged as a part of #2752
